listener.go

```diff
@@ -49,17 +48,21 @@ func (l *onionListener) Addr() net.Addr {
 	return l.addr
 }
 
-// Listener returns a net.Listener backed by a Onion Service, optionally
-// having Tor generate an ephemeral private key.  Regardless of the status of
-// the returned Listener, the Onion Service will be torn down when the control
-// connection is closed.
-//
-// WARNING: Only one port can be listened to per PrivateKey if this interface
-// is used.  To bind to more ports, use the  AddOnion call directly.
-func (c *Conn) Listener(port uint16, key crypto.PrivateKey) (net.Listener, error) {
-	const (
-		loopbackAddr = "127.0.0.1:0"
-	)
+// Listener returns a net.Listener backed by an Onion Service using configuration
+// cfg, optionally having Tor generate an ephemeral private key (cfg in nil or
+// cfg.PrivateKey is nil).
+// All of virtual ports specified in vports will be mapped to the port to which
+// the underlying TCP listener binded.
+// Regardless of the status of the returned Listener, the Onion Service will be
+// torn down when the control connection is closed.
```

Comment:
The Onion Service can persist past the control port connection depending on the config with these changes, update the comment accordingly.